formats/gltf/writer.go

```diff
@@ -496,11 +496,43 @@ func (w *Writer) AddMesh(model PolyformModel) (_ int, err error) {
 }
 
 func (w *Writer) AddTexture(polyTex PolyformTexture) *TextureInfo {
-	newTexInfo := &TextureInfo{Index: len(w.textures)}
+	var imageIndex, samplerIndex int
+	{
+		imageIndex = len(w.images)
+		var imageFound bool
+		for i, im := range w.images {
+			if im.URI == polyTex.URI {
+				imageIndex = i
+				imageFound = true
+				break
+			}
+		}
+		if !imageFound {
+			w.images = append(w.images, Image{URI: polyTex.URI})
+		}
+	}
 
-	newTex := Texture{
-		Sampler: ptrI(len(w.samplers)),
-		Source:  ptrI(len(w.images)),
+	{
+		var texSampler Sampler
+		if polyTex.Sampler != nil {
+			texSampler = *polyTex.Sampler
+		}
+
+		samplerIndex = len(w.samplers)
+		var samplerFound bool
+		for i, sam := range w.samplers {
+			if sam.MagFilter == texSampler.MagFilter &&
+				sam.MinFilter == texSampler.MinFilter &&
+				sam.WrapS == texSampler.WrapS &&
+				sam.WrapT == texSampler.WrapT {
+				samplerIndex = i
+				samplerFound = true
+				break
+			}
+		}
```

Comment:
I think this is a valid spot to re-use the same logic we did with Meshes where we keep up with the pointers of the samplers and check the addresses instead of the values of the samplers themselves, with extra logic around nil samplers.

Takes care of usecases where a user has special logic around sampler names or custom sampler extensions. Idk who would do that, but /shrug

Speaking of nil samplers, I was totally wrong, it's valid for textures to not have samplers. Sorry about that.
* https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#textures
* https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#_texture_sampler

Reply:
Cool, yes, I've converted to pointer comparison. Also removed default empty sampler.